### PR TITLE
Transforms: Add Alpha Format Time Transform

### PR DESCRIPTION
**Pull request by codeincarnate**
**What is this feature?**

This PR adds a straightforward transform to format time fields using a Moment.js formatting string.

**Why do we need this feature?**

This makes is straightforward to format dates for display. It also allows for grouping by day, week, month and year when used in conjunction with the Group by Value transform.

### CODE REVIEW

**Comment by codeincarnate** [comment]
Not sure why this keeps getting marked `pr/external` 🤔

**Comment by leeoniya** [comment]
> This makes is straightforward to format dates for display.

i think this already exists by setting the `unit` on field config to `time:YYYY-MM-DD`

File: packages/grafana-data/src/transformations/transformers/formatTime.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ */
+export const createTimeFormatter =
+  (timeField: string, outputFormat: string) => (fields: Field[], data: DataFrame[], frame: DataFrame) => {
+    return fields.map((field) => {
+      // Find the configured field
+      if (field.name === timeField) {
+        // Update values to use the configured format
+        const newVals = field.values.map((value) => {
+          const moment = dateTime(value);
+          return moment.format(outputFormat);
```
Comment (mdvictor) [inline packages/grafana-data/src/transformations/transformers/formatTime.ts:49]: I think the timezone should also be applied. If the timerange is set to anything other than the local browser tz, the formatted output is different that the initial input

**Review by mdvictor — CHANGES_REQUESTED** [review]
Looks good overall! Need to only fix the TZ issue.

I didn't know that something similar existed already through standard options though, learned something new. The two functionalities seems to be functioning quite the same. Would this still make sense under these circumstances?

I think these custom units are also rather hard to discover for a user. Even though they are specified in the [docs](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-standard-options/), there are no clues about it in the UI

**Comment by codeincarnate** [comment]
@leeoniya first sir, think of the children! 😂 

> i think this already exists by setting the unit on field config to time:YYYY-MM-DD

This does exist but won't work in conjunction with other transforms 😢 so things like group by value won't work in that case

File: packages/grafana-data/src/transformations/transformers/formatTime.ts (hunk 1 above)
Comment (codeincarnate) [reply packages/grafana-data/src/transformations/transformers/formatTime.ts:49]: Good catch! 💟

**Review by mdvictor — APPROVED** [review]

**Review by baldm0mma — APPROVED** [review]
LGTM! I added the "no-backport" label, since I assume we're not back porting this. :)

**Pull request merged**